codersdk/workspaces.go

```diff
@@ -316,10 +316,31 @@ func (c *Client) PutExtendWorkspace(ctx context.Context, id uuid.UUID, req PutEx
 	return nil
 }
 
+type PostWorkspaceUsageRequest struct {
+	AgentID uuid.UUID    `json:"agent_id" format:"uuid"`
+	AppName UsageAppName `json:"app_name"`
+}
+
+type UsageAppName string
+
+const (
+	UsageAppNameVscode          UsageAppName = "vscode"
+	UsageAppNameJetbrains       UsageAppName = "jetbrains"
+	UsageAppNameReconnectingPty UsageAppName = "reconnecting-pty"
+	UsageAppNameSSH             UsageAppName = "ssh"
+)
+
+var AllowedAppNames = []UsageAppName{
+	UsageAppNameVscode,
+	UsageAppNameJetbrains,
+	UsageAppNameReconnectingPty,
+	UsageAppNameSSH,
+}
+
 // PostWorkspaceUsage marks the workspace as having been used recently.
-func (c *Client) PostWorkspaceUsage(ctx context.Context, id uuid.UUID) error {
+func (c *Client) PostWorkspaceUsage(ctx context.Context, id uuid.UUID, req PostWorkspaceUsageRequest) error {
```

Comment:
note: this is technically a breaking change in the public codersdk API; we should either call it out or make this a separate method.

Reply:
I'll do a separate method 👍